include/ibamr/BrinkmanPenalizationRigidBodyDynamics.h

```diff
@@ -0,0 +1,372 @@
+     * \brief Kinematics function data.
+     */
+    struct KinematicsFcnData
+    {
+        KinematicsFcnData(KinematicsFcnPtr comvelfcn = nullptr, void* ctx = nullptr) : comvelfcn(comvelfcn), ctx(ctx)
+        {
+            // intentionally left blank
+        }
+        KinematicsFcnPtr comvelfcn;
+        void* ctx;
+    };
+
+    /*!
+     * \brief Register user defined constrained velocity functions.
+     */
+    void registerKinematicsFunction(KinematicsFcnPtr comvelfcn, void* ctx = nullptr);
+
+    /*!
+     * \brief Register user defined velocity function data.
+     */
+    void registerKinematicsFunction(const KinematicsFcnData& data);
+
+    /*!
+     * \brief Set what rigid DOFs need to be solved for this
+     * particular structure.
+     */
+    virtual void setSolveRigidBodyVelocity(const IBTK::FreeRigidDOFVector& solve_rigid_dofs);
+
+    /*!
+     * \brief Typedef specifying interface for specifying additional rigid body force and torque.
+     */
+    using ExternalForceTorqueFcnPtr = void (*)(double data_time, Eigen::Vector3d& F, Eigen::Vector3d& T, void* ctx);
+
+    /*
+     * \brief External force/torque function data.
+     */
+    struct ExternalForceTorqueFcnData
+    {
+        ExternalForceTorqueFcnData(ExternalForceTorqueFcnPtr forcetorquefcn = nullptr, void* ctx = nullptr)
+            : forcetorquefcn(forcetorquefcn), ctx(ctx)
+        {
+            // intentionally left blank
+        }
+        ExternalForceTorqueFcnPtr forcetorquefcn;
+        void* ctx;
+    };
+
+    /*!
+     * \brief Register user defined external force/torque functions.
+     */
+    void registerExternalForceTorqueFunction(ExternalForceTorqueFcnPtr forcetorquefcn, void* ctx = nullptr);
+
+    /*!
+     * \brief Register user defined external force/torque function data.
+     */
+    void registerExternalForceTorqueFunction(const ExternalForceTorqueFcnData& data);
+
+    /*!
+     * \brief Preprocess routine before computing Brinkman penalization terms.
+     *
+     */
+    void preprocessComputeBrinkmanPenalization(double current_time, double new_time, int num_cycles) override;
+
+    /*!
+     * \brief Compute the desired rigid body velocity in the Brinkman penalized zone.
+     */
+    void computeBrinkmanVelocity(int u_idx, double time, int cycle_num) override;
+
+    /*!
+     * \brief Demarcate the Brinkman zone with Brinkman penalty term.
+     */
+    void demarcateBrinkmanZone(int u_idx, double time, int cycle_num) override;
+
+    /*!
+     * \brief Postprocess routine after computing Brinkman penalization terms.
+     */
+    void postprocessComputeBrinkmanPenalization(double current_time, double new_time, int num_cycles) override;
+
+    /*!
+     * \brief Write out object state to the given database.
+     */
+    void putToDatabase(SAMRAI::tbox::Pointer<SAMRAI::tbox::Database> db) override;
+
+    /*!
+     * \brief Get initial center of mass of the body.
+     */
+    const Eigen::Vector3d& getInitialCOMPosn() const
```

Comment:
Should these be marked as `inline`? (And possibly moved to a `private/BrinkmanPenalizationRigidBodyDynamics-inl.h` file? Although I do not care about this as much as I use to.)

Reply:
Functions defined in the class declaration are implicitly inline so its not needed.